gencode.c

```diff
@@ -661,6 +661,103 @@ syntax(compiler_state_t *cstate)
 }
 
 int
+pcap_is_raw(const char *bpf_string)
+{
+	char sp, separator1 = ',', separator2 = '\n';
+	unsigned short bpf_len;
+
+	if (bpf_string == NULL)
+		return 0;
+
+	if (sscanf(bpf_string, "%hu%c", &bpf_len, &sp) != 2 ||
+			(sp != separator1 && sp != separator2))
+		return 0;
+
+	return 1;
+}
+
+int
+pcap_compile_raw(pcap_t *p, struct bpf_program *program, const char *bpf_string)
+{
+	const char *token;
+	char sp, separator, separator1 = ',', separator2 = '\n';
+	unsigned short bpf_len, i = 0, line = 0;
+	size_t size;
+	struct bpf_insn tmp;
+
+	program->bf_len = 0;
+
+	if (sscanf(bpf_string, "%hu%c", &bpf_len, &sp) != 2 ||
+			(sp != separator1 && sp != separator2) ||
+			bpf_len > BPF_MAXINSNS || bpf_len == 0) {
```

Comment:
Note that my kernel (3.13.0) defines BPF_MAXINSNS to be 4096. I doubt there is an issue with moving from 4096 to 64k (raw filters are only used from tcpdump clients).

Another question: do you prefer me harcoding the 64k constant, or something like:

#ifndef BPF_MAXINSNS
#define BPF_MAXINSNS 65536
#endif